Task4.1/DiagonalMatrixExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace MatrixTask
+{
+    public static class DiagonalMatrixExtensions
+    {
+        public static DiagonalMatrix<T> Add<T>(this DiagonalMatrix<T> matrix1, DiagonalMatrix<T> matrix2, Func<T, T, T> addFunction)
+        {
+            int maxSize = Math.Max(matrix1.Size, matrix2.Size);
+            DiagonalMatrix<T> resultMatrix = new DiagonalMatrix<T>(maxSize);
+
+            for (int i = 0; i < maxSize; i++)
+            {
+                T value1 = matrix1[i, i];
+                T value2 = matrix2[i, i];

```

Comment:
Can be simplified:
1. resultMatrix[i, i] = addFunction(matrix1[i,i], matrix2[i,i]);
2. It seems that it will be an exception if matrix1.Size != matrix2.Size